developer/decision-records/2024-07-17-bootstrap-sql-database/README.md

```diff
@@ -0,0 +1,33 @@
+# Bootstrapping SQL databases
+
+## Decision
+
+The EDC project will provide functionality to automatically create database schemas for every SQL store module. This facility will be available in all components that have SQL persistence and can be globally enabled or disabled. It will be set to _disabled_ by default. 
+
+Note that this is not a fully-fledged migration solution, as it will not provide backwards compatibility or data migration.
+
+## Rationale
+
+Currently, users of EDC have to come up with their very own way to create database schemas whenever they deploy a connector. While this may be acceptable (and even desired) in large-scale production environments, where dedicated database engineering teams are available that also perform data migration, it creates a considerable barrier of entry for new adopters of EDC. 
+To even spin up a demo or a proof-of-concept connector that uses PostgreSQL, developers have to also find a way to create the database structure. While it is not particularly hard to do, it needlessly makes life harder in these situations. 
+
+As an option, EDC will offer a config switch `edc.database.schema.autocreate=true|false`, that globally enables or disables the automatic bootstrapping of the database.
```

Comment:
I would call the setting `edc.sql.schema.autocreate`, because `database` is too generic, and we already have `edc.sql....` settings